components/reusable-components/newsletter/emailform/emailform.tsx

```diff
@@ -0,0 +1,67 @@
+'use client';
+
+import React, { useState } from 'react';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import Notification from './notification/notification';
+import Style from './notification/notification.module.scss';
+
+interface EmailFormProps {
+  inputClassName?: string;
+  buttonClassName?: string;
+}
+
+const EmailForm: React.FC<EmailFormProps> = ({ inputClassName, buttonClassName }) => {
+  const [showNotification, setShowNotification] = useState(false);
+  const [notificationMessage, setNotificationMessage] = useState('');
+  const [notificationType, setNotificationType] = useState<'success' | 'error'>('success');
+
+  const formik = useFormik({
+    initialValues: { email: '' },
+    validationSchema: Yup.object({
+      email: Yup.string().email('Invalid email address!').required('Email is required!'),
```

Comment:
you will need a custom regex here since this validation won't be enough.
and then do .matches with that regex